### PR TITLE
[ssl-certificate-expiry] host name verification + proxy connection

### PR DESCRIPTION
**Pull request by Nazdravi**
Optional verification of request to certificate hostname match (env.checkname yes)
Optional openssl proxy usage (env.proxy PROXYHOST:PORT)

### CODE REVIEW

File: plugins/ssl/ssl-certificate-expiry [hunk 1]
```diff
@@ -139,10 +144,15 @@ print_expire_days() {
...
-    echo "" | openssl s_client \
-	-servername "$host" -connect "${host}:${port}" \
-	-showcerts \
-	$s_client_args 2>/dev/null | \
+    local openssl_call
+    local openssl_response
+    openssl_call="s_client -servername $host -connect ${host}:${port} -showcerts $s_client_args"
+    # shellcheck disable=SC2086
+    openssl_response=$(echo "" | openssl $openssl_call 2>/dev/null)
+    if [ "$(echo "$openssl_response" | grep -i -c "Hostname mismatch")" -gt "0" ]; then
```
Comment (sumpfralle) [inline plugins/ssl/ssl-certificate-expiry:152]: Maybe simplify this a bit?

```shell
if echo "$openssl_response" | grep -qi "Hostname mismatch"; then
```

File: plugins/ssl/ssl-certificate-expiry [hunk 2]
```diff
@@ -168,9 +178,12 @@ main() {
 	fi
 	fieldname="$(clean_fieldname "$service")"
 	valid_days=$(print_expire_days "$host" "$port" "$starttls")
+	extinfo=""
 	[ -z "$valid_days" ] && valid_days="U"
+	[ "$valid_days" = "<>" ] && extinfo="Error: hostname mismatch, "
```
Comment (sumpfralle) [inline plugins/ssl/ssl-certificate-expiry:183]: I think, a real `if ... then; ... fi` statement would increase readability here.

File: plugins/ssl/ssl-certificate-expiry [hunk 3]
```diff
@@ -168,9 +178,12 @@ main() {
...
+	[ "$valid_days" = "<>" ] && valid_days="-1"
 	printf "%s.value %s\\n" "$fieldname" "$valid_days"
-        echo "${fieldname}.extinfo Last checked: $(date)"
+        echo "${fieldname}.extinfo ${extinfo}Last checked: $(date)"
```
Comment (sumpfralle) [inline plugins/ssl/ssl-certificate-expiry:186]: Here you could fix the indentation (to match the lines above).

**Review by sumpfralle — COMMENTED** [review]
Thank you for your contribution!

Please take a look at my minor comments below.

**Comment by sumpfralle** [comment]
Thank you for your recent changes.
From my point of view it is ready to be merged.
Or do you plan any further additions?

**Comment by Nazdravi** [comment]
Thanks for your comments. From my side this is finished for now and I'm not planning further changes.

Hi Lars, klein die Welt ;-)

**Comment by sumpfralle** [comment]
Thanks for your contribution!

> Hi Lars, klein die Welt ;-)

So sieht es aus! Wobei das Erkennen tatsächlich aufgrund deiner schmalen Datenspuren gerade nur einseitiger Natur ist :)

**Pull request merged**